src/tracks/timetrack/ui/TimeTrackView.cpp

```diff
@@ -156,9 +158,26 @@ void TimeTrackView::Draw(
    const wxRect &rect, unsigned iPass )
 {
    if ( iPass == TrackArtist::PassTracks ) {
+      const auto pTrack = FindTrack();
+      auto pList = pTrack->GetOwner();
+      if (!pList)
+         // Track isn't owned by a list.  Can't proceed!
+         return;
+      if (!pList)
+         // List isn't owned by a project.  Can't proceed!
+         // But this shouldn't happen when drawing it
+         return;
+
+      auto pProject = pList->GetOwner();
+      auto &zoomInfo = ViewInfo::Get(*pProject);
+      LinearUpdater updater;
+      updater.SetData(&zoomInfo);
+      Ruler ruler{ updater, TimeFormat::Instance() };
+      ruler.SetLabelEdges(false);
```

Comment:
I see that `Ruler` does some cacheing, wouldn't it be better to make `LinearUpdater` and `Ruler` a member variable of `TimeTrackView`?

Reply:
I debated this with myself.  But I think this ruler isn't saving any important state and the calculations are not large.  I may change my mind.